examples/md-flexible/src/Simulation.cpp

```diff
@@ -372,6 +465,106 @@ void Simulation::calculateGlobalForces(const std::array<double, 3> &globalForce)
   }
 }
 
+void Simulation::logSimulationState() {
+  int totalNumberOfParticles, ownedParticles, haloParticles;
```

Comment:
```suggestion
  size_t totalNumberOfParticles{0ul}, ownedParticles{0ul}, haloParticles{0ul};
```
Particle numbers might not fit in `int`. You will also need to adapt the Allreduce calls below.